src/sgl/func/tensor.cpp

```
@@ -0,0 +1,222 @@
+        if (!dtype)
+            SGL_THROW("No derivative type found for {}", m_desc.dtype->name());
+        ref<TypeLayoutReflection> layout = dtype->buffer_type_layout();
+
+        BufferDesc buffer_desc;
+        buffer_desc.usage = BufferUsage::shader_resource | BufferUsage::unordered_access | BufferUsage::shared;
+        buffer_desc.struct_size = layout->stride();
+        buffer_desc.element_count = element_count();
+        ref<Buffer> buffer = device()->create_buffer(buffer_desc);
+
+        TensorDesc desc;
+        desc.dtype = dtype;
```

Cppcheck (2.20.0)

[style] The function 'info' is never used.

(unusedFunction)